static/settings.ts

```diff
@@ -265,13 +273,14 @@ export class Settings {
     }
 
     private addSelectors() {
-        const addSelector = (
+        const addSelector = <Name extends keyof SiteSettings>(
             selector: string,
-            name: keyof SiteSettings,
+            name: Name,
             populate: {label: string; desc: string}[],
-            defaultValue: string
+            defaultValue: SiteSettings[Name],
+            component = Select
```

Comment:
Great stuff, love the use of the Name generic here

Reply:
Thanks to Ruben for showing me how to make this work!